Content.Server/GameObjects/Components/Items/rcdComponent.cs

```diff
@@ -0,0 +1,166 @@
+
+        private enum RCDmodes //Enum to store the different mode states for clarity.
+        {
+            Floors,
+            Walls,
+            Deconstruct
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _outputTile, "output", "floor_steel");
+        }
+
+        /**
+         *
+         *    Method called when the RCD is clicked in-hand, this will swap the RCD's mode from "floors" to "walls".
+         *
+         */
+
+        public bool UseEntity(UseEntityEventArgs eventArgs)
+        {
+            SwapMode(eventArgs);
+            return false;
+        }
+
+
+        /**
+         *
+         * Un-used method which is inherited from super
+         *
+         */
+        public void Activate(ActivateEventArgs eventArgs)
+        {
+            return;
+        }
+
+        /**
+         *
+         * Method to allow the user to swap the mode of the RCD by clicking it in hand, the actual in hand clicking bit is done over on UseEntity()
+         *
+         * @param UseEntityEventArgs = The entity which triggered this method call, used to know where to play the "click" sound.
+         *
+         */
+
+        public void SwapMode(UseEntityEventArgs eventArgs)
+        {
+            _entitySystemManager.GetEntitySystem<AudioSystem>().Play("/Audio/items/genhit.ogg", Owner);
+            this.mode = (this.mode < modes.Length-1) ? this.mode + 1 : 0; //Basic value rollover so that you can't get to an invalid mode
+            string mode = modes[this.mode];
+            switch (mode)
+            {
+                case("Floors"):
+                    _outputTile = "floor_steel";
+                    break;
+                case("Walls"):
+                    _outputTile = "base_wall";
+                    break;
+                case("Deconstruct"):
+                    _outputTile = "space";
+                    break;
+            }
+            var notify = IoCManager.Resolve<IServerNotifyManager>();
+            notify.PopupMessage(Owner, eventArgs.User, "The RCD will now place "+mode); //Prints an overhead message above the RCD
+        }
+
+        /**
+         *
+         * Method called when the user examines this object, it'll simply add the mode that it's in to the object's description
+         * @params message = The original message from examining, like ..() in BYOND's examine
+         *
+         */
+
+        void IExamine.Examine(FormattedMessage message)
+        {
+            var loc = IoCManager.Resolve<ILocalizationManager>();
+            string mode = Enum.GetNames(typeof(RCDmodes))[this.mode]; //Access the string name of the mode based off of its numerical index in the enum.
+            message.AddMarkup(loc.GetString("It's currently placing "+mode+", and holds "+this.ammo+" charges."));
```

Comment:
Use [string interpolation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) here instead.

Reply:
I disagree. This should use the string formatting, to avoid having to define a translation string for every mod and every number of charges:

```c#
Loc.GetString("It's currently placing {0}, and holds {1} charges.", Loc.GetParticularString("RCDMode", mode), this.ammo);
```

Reply:
Sounds like Loc.GetString needs to be updated to handle FormattableString types properly...

Reply:
Sounds like Loc.GetString needs to be updated to handle FormattableString types properly...